src/tests/tools/iotc_test_echoserver.cc

```diff
@@ -0,0 +1,164 @@
+/* Copyright 2018-2019 Google LLC
+ *
+ * This is part of the Google Cloud IoT Device SDK for Embedded C,
+ * it is licensed under the BSD 3-Clause license; you may not use this file
+ * except in compliance with the License.
+ *
+ * You may obtain a copy of the License at:
+ *  https://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "iotc_test_echoserver.h"
+
+#include <netinet/in.h>
+#include <string.h>
+#include <thread>
+
+namespace iotctest {
+
+EchoTestServer::EchoTestServer(uint16_t socket_type, uint16_t port,
```

Comment:
This is a weird sequence of parameters.

`socket_type, protocol_type, host, port` ?
or
`host, port, socket_type, protocol_type` ?